java/core/src/main/java/co/worklytics/psoxy/storage/impl/CSVFileHandler.java

```diff
@@ -0,0 +1,100 @@
+
+@NoArgsConstructor(onConstructor_ = @Inject)
+public class CSVFileHandler implements FileHandler {
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Override
+    public byte[] handle(@NonNull InputStreamReader reader, @NonNull Sanitizer sanitizer) throws IOException {
+        CSVParser records = CSVFormat.DEFAULT
+                .withFirstRecordAsHeader()
+                .parse(reader);
+
+        Preconditions.checkArgument(records.getHeaderMap() != null, "Failed to parse header from file");
+
+        Map<Integer, String> invertedHeaderMap = records.getHeaderMap().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey, (a, b) -> a, TreeMap::new));
+
+        Sanitizer.Options options = sanitizer.getOptions();
+
+        Set<String> columnsToRedact = options.getRules()
+                .getRedactions()
+                .stream()
+                .map(Rules.Rule::getColumns)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+
+        Set<String> columnsToPseudonymize = options.getRules()
+                .getPseudonymizations()
+                .stream()
+                .map(Rules.Rule::getColumns)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+
+        String[] header = invertedHeaderMap.values()
+                .stream()
+                .filter(entry -> !columnsToRedact.contains(entry)).toArray(String[]::new);
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream(1024);
+        PrintWriter printWriter = new PrintWriter(baos);
+        CSVPrinter printer = new CSVPrinter(printWriter, CSVFormat.DEFAULT.withHeader(header));
+
+        columnsToPseudonymize
+                .forEach(columnToPseudonymize ->
+                        Preconditions.checkArgument(records.getHeaderMap().containsKey(columnToPseudonymize), "Column %s to be pseudonymized not in file", columnToPseudonymize));
+
+        records.forEach(row -> {
+            List<Object> sanitized = invertedHeaderMap.entrySet()
+                    .stream()
+                    .map(column -> {
+                        String value = row.get(column.getKey());
+                        if (columnsToRedact.contains(column.getValue())) {
+                            return null;
+                        } else if (StringUtils.isNotBlank(value) && columnsToPseudonymize.contains(column.getValue())) {
+                            try {
+                                return objectMapper.writeValueAsString(sanitizer.pseudonymize(value));
+                            } catch (JsonProcessingException e) {
+                                throw new RuntimeException(e);
+                            }
+                        } else {
+                            return value;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
```

Comment:
Simplify just iterating the allowed headers. The library also allows to read by column, so no need to use the map.
Also, the filter non null made me a bit nervous... what if a *real* value is null, will be out, thus shifting the mapped columns.
On the tests please check if entries like ",,,,," are handled properly

```suggestion
            List<Object> sanitized = Arrays.stream(header). // only iterate on allowed headers
                    .stream()
                    .map(column -> {
                        String value = row.get(column);
                        if (columnsToPseudonymize.contains(column)) {
                          if (StringUtils.isNotBlank(value)) {
                            try {
                                  return objectMapper.writeValueAsString(sanitizer.pseudonymize(value));
                              } catch (JsonProcessingException e) {
                                  throw new RuntimeException(e);
                              }
                          }
                        }
                        return value;
                    })
                    .collect(Collectors.toList());
```